src/modules/content-s3/content-s3.c

```diff
@@ -11,22 +11,196 @@
+        goto error;
+    }
+
+    if (!(cpy = strdup (uri)))
+        goto error;
+
+    if (yuarel_parse (&yuri, (char *) cpy) < 0) {
+        snprintf(errbuff, eb_size, "failed to parse uri");
+        goto error;
+    }
+
+    ctx->cfg->hostname = yuri.host;
+    if (!(strncat (ctx->cfg->hostname, ":", sizeof(ctx->cfg->hostname) + 1))) {
+        snprintf(errbuff, eb_size, "failed to form hostname");
+        goto error;
+    }
+
+    sprintf(tmp, "%d", yuri.port);
+    if (!(strncat (ctx->cfg->hostname, tmp, sizeof(ctx->cfg->hostname) + sizeof(yuri.port)))) {
+        snprintf(errbuff, eb_size, "failed to form hostname");
+        goto error;
+    }
+
+    if (!(cpy = strdup (bucket)))
+        goto error;
+
+    if (!(ctx->cfg->bucket = strdup (bucket)))
+        goto error;
+
+    if (!strncmp (yuri.scheme, "https", 5))
+        ctx->is_secure = 1;
+
+    if (!(tbl = tomltk_parse_file (cred_file, &toml_error))) {
+        snprintf(errbuff, eb_size, "failed to parse credentials file");
+        goto error;
+    }
+
+    if (!(raw = toml_raw_in(tbl, "secret-access-key"))) {
+        snprintf(errbuff, eb_size, "failed to parse secret key");
+        goto error;
+    }
+
+    if (toml_rtos (raw, &secret_key)) {
+        snprintf(errbuff, eb_size, "failed to parse secret key");
+        goto error;
+    }
+
+    if (!(raw = toml_raw_in(tbl, "access-key-id"))) {
+        snprintf(errbuff, eb_size, "failed to parse access key");
+        goto error;
+    }
+
+    if (toml_rtos (raw, &access_key)) {
+        snprintf(errbuff, eb_size, "failed to parse access key");
+        goto error;
+    }
+
+    ctx->cfg->secret_key = secret_key;
+    ctx->cfg->access_key = access_key;
+
+    return 0;
+
+error:
+    return -1;
+}
+
+/* Broker is sending us a new config object because 'flux config reload'
+ * was run.  Parse it and respond with human readable errors.
+ * If events are posted, block until they complete so that:
+ * - any KVS commit errors are captured by 'flux config reload'
+ * - tests can look for eventlog entry after running 'flux config reload'
+ */
+static void config_reload_cb (flux_t *h,
+                              flux_msg_handler_t *mh,
+                              const flux_msg_t *msg,
+                              void *arg)
+{
+    (void) arg;
+    struct content_s3 *ctx;
+    const flux_conf_t *conf;
+    char errbuf[256];
+    const char *errstr = NULL;
+
+    if (!(ctx = calloc (1, sizeof (*ctx))))
+        goto error;
+    if (!(ctx->cfg = calloc (1, sizeof (*ctx->cfg))))
+        goto error;
+    
+    ctx->is_secure = 0;
+    ctx->is_virtual_host = 0;
+    ctx->cfg->retries = 5;
+    ctx->cfg->bucket = NULL;
+    ctx->cfg->access_key = NULL;
+    ctx->cfg->secret_key = NULL;
+    ctx->cfg->hostname = NULL;
+
+    if (flux_conf_reload_decode (msg, &conf) < 0)
+        goto error;
+
+    flux_log(h, LOG_WARNING, "config relod detectedin content-s3: changes will not take effect until next flux restart");
```

Comment:
spelling errors, and this notification could be moved to after `parse_config()` succeeds.

Since the logs include the module name you might just shorten to something like:
> config-reload: changes will not take effect until flux is restarted